service/frontend/adminHandler.go

```diff
@@ -85,53 +77,29 @@ type (
 
 // NewAdminHandler creates a thrift handler for the cadence admin service
 func NewAdminHandler(
-	sVice service.Service,
-	numberOfHistoryShards int,
-	domainCache cache.DomainCache,
-	historyV2Mgr persistence.HistoryManager,
+	resource resource.Resource,
 	params *service.BootstrapParams,
 	config *Config,
 ) *AdminHandler {
-	handler := &AdminHandler{
-		status:                common.DaemonStatusInitialized,
-		numberOfHistoryShards: numberOfHistoryShards,
-		Service:               sVice,
-		domainCache:           domainCache,
-		historyV2Mgr:          historyV2Mgr,
+	return &AdminHandler{
+		Resource:              resource,
+		numberOfHistoryShards: params.PersistenceConfig.NumHistoryShards,
```

Comment:
Why both bootstrapParams and resource have the field for number of shards? Which one should be used?

Reply:
The one in resource is just an internal field and can't be retrieved. I can delete it if it causes any confusion.